### ISSUE TITLE
Diff for applications with multiple sources

### ISSUE THREAD

**Issue by andridzi** [zapier/kubechecks#294]
Hello,

It would be greatly appreciated to have this feature, as the current implementation doesn't seem to work for applications with multiple sources (like Helm deployments, where one source is for the Helm chart itself and another for the values).

I noticed an initial effort from @sl1pm4t in #240, but it appears to still be in draft.

**Comment by memdealer** [comment]
YES PLEASE.

**Comment by alaa-404** [comment]
would be gr8 💯

**Comment by djeebus** [comment]
For some reason I remember this working, but that doesn't seem to be true. OK, I built on the work that @sl1pm4t did and got a working branch. I don't have any multi source applications to test on yet, but if anyone wants a sneak preview, you can use the image at `ghcr.io/zapier/kubechecks:0.0.0-pr298`, and report back if it works as intended.

**Comment by andridzi** [comment]
@djeebus, I'll test and comment

**Comment by andridzi** [comment]
@djeebus, so, it's not crashing anymore but still not working
current setup

```
    sources:
      - repoURL: https://charts.fairwinds.com/stable
        chart: rbac-manager
        targetRevision: 1.6.0
        helm:
          releaseName: rbac-manager
          valueFiles:
          - '$values/{{metadata.labels.environment}}/rbac-manager/{{name}}.yaml'
      - repoURL: https://github.com/andridzi/helm-values.git
        targetRevision: master
        ref: values
```
first error - it failed to clone `https://stakater.github.io/stakater-charts` repo
```
3:56PM ERR Unable to clone repository error="failed to clone repo: failed to clone repository: exit status 128" app_name=rbac-manager app_path= event_id=476 repo=argocd workerID=0
3:56PM INF cloning git repo branch=1.6.0 clone-url=https://andridzi@charts.fairwinds.com/stable temp-dir=/tmp/kubechecks-repo-1262552353
```

after changing git ref to 
```
      sources:
        repoURL: git@github.com:FairwindsOps/charts.git
        path: stable/rbac-manager
```

it failed with error that it couldn't find the values file

**Comment by djeebus** [comment]
Teach me to dive in without understanding what I was doing. `ref: values` is a new feature to me. I'm going to do some reading about this, see what I'm missing, and how much I need to update to push this stuff back onto argocd code. Stay tuned!

**Comment by sl1pm4t** [comment]
For the record we use the `ref: values` pattern heavily too.

**Comment by andridzi** [comment]
@djeebus 
So, it's not crashing now :)

1. Need to add possibility to specify `command` and `args` (not only `args`) to be able to setup custom endpoint for ArgoCD server
2. `kubechecks-rbac` chart will be obsolete if move RBAC part into here (which I prefer actually)
3. Still cannot specify repo URL in `https` format, like:
```
      - repoURL: https://charts.fairwinds.com/stable
        chart: rbac-manager
        targetRevision: 1.6.0
```
when doing like that - getting error about cloning the repository
using `git@github.com` will not work, because we need to select exact chart version (tag), not a branch, which can be non-existing at all

4. When speifying multiple repositories we suggest files to be changed there as well (like values for helm charts or additional manifests to be applied).
But currently when I'm doing some changes in these repositories it's just showing `INF No affected apps or appsets, skipping`.
To be able to use this tools for multi-source repos - it should check (I don't know how for now), if this particular file (values of manifest) is used in some ArgoCD app/appset or no.

P.S. Big thanks for working on it, the tool is really great!

**Comment by anthonyhaussman** [comment]
Also using the `ref: values` pattern a lot with Helm repositories `repoURL`.
It would be a big game changer when it'll work. :heart:

**Comment by djeebus** [comment]
This feature should now (mostly) work! It was released yesterday as v2.0.1 (after fixing a last minute bug). The one thing still doesn't work is referencing helm charts directly, that'll be a follow up feature.

Enjoy!

@andridzi you should be able to change the internal argocd endpoint via the `KUBECHECKS_ARGOCD_REPOSITORY_ENDPOINT` env var. You can already change the external argocd endpoint via `KUBECHECKS_ARGOCD_API_SERVER_ADDR`. As for your last point, I think it _should_ work with multiple sources now. 🤞 Mind trying the newly released version?

**Comment by djeebus** [comment]
I'm going to close this issue, as the feature has been implemented, but if you find any bugs, please open a new issue! Thanks!
